pkg/artifactory/resource/security/resource_artifactory_scoped_token_test.go

```diff
@@ -0,0 +1,229 @@
+			},
+		},
+	})
+}
+
+func TestAccScopedToken_WithAttributes(t *testing.T) {
+	_, fqrn, name := acctest.MkNames("test-access-token", "artifactory_scoped_token")
+
+	accessTokenConfig := acctest.ExecuteTemplate("TestAccScopedToken", `
+		resource "artifactory_user" "test-user" {
+			name              = "testuser"
+		    email             = "testuser@tempurl.org"
+			admin             = true
+			disable_ui_access = false
+			groups            = ["readers"]
+			password          = "Passw0rd!"
+		}
+
+		resource "artifactory_scoped_token" "{{ .name }}" {
+			username    = artifactory_user.test-user.name
+			scopes      = ["applied-permissions/admin", "system:metrics:r"]
+			description = "test description"
+			refreshable = true
+			audiences   = ["admin"]
+		}
+	`, map[string]interface{}{
+		"name": name,
+	})
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ProviderFactories: acctest.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: accessTokenConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "username", "testuser"),
+					resource.TestCheckResourceAttr(fqrn, "scopes.#", "2"),
+					resource.TestCheckTypeSetElemAttr(fqrn, "scopes.*", "applied-permissions/admin"),
+					resource.TestCheckTypeSetElemAttr(fqrn, "scopes.*", "system:metrics:r"),
+					resource.TestCheckResourceAttr(fqrn, "refreshable", "true"),
+					resource.TestCheckResourceAttr(fqrn, "description", "test description"),
+					resource.TestCheckResourceAttr(fqrn, "audiences.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "audiences.0", "admin"),
+					resource.TestCheckResourceAttrSet(fqrn, "access_token"),
+					resource.TestCheckResourceAttr(fqrn, "token_type", "Bearer"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScopedToken_WithInvalidScopes(t *testing.T) {
+	_, _, name := acctest.MkNames("test-scoped-token", "artifactory_scoped_token")
+
+	scopedTokenConfig := acctest.ExecuteTemplate("TestAccScopedToken", `
+		resource "artifactory_local_generic_repository" "generic-local-1" {
+			key = "generic-local-1"
+		}
+
+		resource "artifactory_local_generic_repository" "generic-local-2" {
+			key = "generic-local-2"
+		}
+
+		resource "artifactory_local_generic_repository" "generic-local-3" {
+			key = "generic-local-3"
+		}
+
+		resource "artifactory_local_generic_repository" "generic-local-4" {
+			key = "generic-local-4"
+		}
+
+		resource "artifactory_user" "test-user" {
+			name              = "testuser"
+		    email             = "testuser@tempurl.org"
+			admin             = true
+			disable_ui_access = false
+			groups            = ["readers"]
+			password          = "Passw0rd!"
+		}
+
+		resource "artifactory_scoped_token" "{{ .name }}" {
+			username    = artifactory_user.test-user.name
+			scopes      = [
+				"applied-permissions/admin",
+				"applied-permissions/user",
+				"system:metrics:r",
+				"system:livelogs:r",
+				"artifact:generic-local-1:r",
```

Comment:
So, the first part of the scope can be:
- artifact
- system
- `foo/baz`
?

Reply:
For the `:` separated value, it's either `system` or `artifact`. The doc/description mentions this but it isn't very obvious.